### PR TITLE
TD-2508- Courses that are marked as hidden in learning portal should not appear in the Available Activities list

### PR DESCRIPTION
**Pull request by Auldrin-Possa**
**JIRA link**
https://hee-tis.atlassian.net/browse/TD-2508

**Description**
Stored procedure modified to get all records which are not hidden in learning portal.

**Screenshots**
image

image

-----
**Developer checks**

I have:
- [ ] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence]
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

**Comment by kevwhitt-hee** [comment]
@Auldrin-Possa, have you checked that the stored procedure is not being called from elsewhere in the application. In particular, Tracking System -> Delegates -> Manage Delegate -> Enrol on activity pulls a list of available activities and the ones that are marked as hidden in learning portal should appear here.

**Comment by Auldrin-Possa** [comment]
> @Auldrin-Possa, have you checked that the stored procedure is not being called from elsewhere in the application. In particular, Tracking System -> Delegates -> Manage Delegate -> Enrol on activity pulls a list of available activities and the ones that are marked as hidden in learning portal should appear here.

@kevwhitt-hee, I have made changes to stored proc accordingly. Please have a look.

File: DigitalLearningSolutions.Data/Models/Courses/AvailableCourse.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@ public class AvailableCourse : BaseLearningItem
             set => topic = GetValidOrNull(value);
         }
         public int DelegateStatus { get; set; }
+        public int HideInLearnerPortal { get; set; }
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Data/Models/Courses/AvailableCourse.cs:19]: This should be a boolean, I think?

File: DigitalLearningSolutions.Web.Tests/TestHelpers/AvailableCourseHelper.cs [hunk 2]
```diff
@@ -16,7 +16,8 @@ public static class AvailableCourseHelper
             string brand = "Brand 1",
             string? category = "Category 1",
             string? topic = "Topic 1",
-            int delegateStatus = 0
+            int delegateStatus = 0,
+            int hideInLearnerPortal =0
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web.Tests/TestHelpers/AvailableCourseHelper.cs:20]: The VS auto formatter should have added a space after the = here. Please could you run autoformat on any files changed?

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
Looks good. A couple of very minor things to tidy up.

**Comment by Auldrin-Possa** [comment]
> Looks good. A couple of very minor things to tidy up.

variable type modified.

**Pull request merged**